### ISSUE TITLE
Can't open Add link in Navigation Submenu via Enter Keypress

### ISSUE THREAD

**Issue by jeryj** [WordPress/gutenberg#60051]
### Description

When "Add link" is selected in a navigation submenu, pressing Enter should open the link control for that navigation item. 

There is no way to prevent the [`insertAfterBlock` from running on an Enter keypress](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/use-block-props/use-selected-block-event-handlers.js#L59-L60).

### Step-by-step reproduction instructions

- Create a navigation menu
- Add a submenu
- Use the arrow keys to navigate to the Add link (it will have a blue border around it)
- Press Enter
- A new Add Link will be added with the link control open

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/967608/55eb187c-e6fe-49c4-9753-30180d9b4051

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jordesign** [comment]
Thanks for the fix @jeryj  - does that mean this issue can be closed?

**Comment by jeryj** [comment]
No, sorry. This issue is not fixed, so this needs to stay open. I was linking to another PR that introduced the behavior. There is not a way to override the behavior of Enter keypress on a block adding a new block if there is a default block provided.
